datastream-kafka-connector/src/test/java/com/linkedin/datastream/connectors/kafka/mirrormaker/TestKafkaMirrorMakerConnectorTask.java

```diff
@@ -325,6 +333,109 @@ public void testPartitionManagedLockReleaseOnConsumerCloseException() throws Exc
     Assert.assertTrue(releaseCall.await(10, TimeUnit.SECONDS), "DatastreamTask never released");
   }
 
+  static class KafkaMirrorMakerConnectorTaskTest extends KafkaMirrorMakerConnectorTask {
+    boolean postShutdownHookExceptionCaught;
+
+    /**
+     * Constructor for TestInterruptKafkaMirrorMakerConnectorTask
+     * @param config Task configuration properties
+     * @param task Datastream task
+     * @param connectorName Connector name
+     * @param isFlushlessModeEnabled true if {@value KafkaMirrorMakerConnector#IS_FLUSHLESS_MODE_ENABLED}
+     *                               is set to true for the connector identified with {@code connectorName}
+     * @param groupIdConstructor Kafka consumer group ID constructor
+     */
+    public KafkaMirrorMakerConnectorTaskTest(KafkaBasedConnectorConfig config, DatastreamTask task,
+        String connectorName, boolean isFlushlessModeEnabled, GroupIdConstructor groupIdConstructor) {
+      super(config, task, connectorName, isFlushlessModeEnabled, groupIdConstructor);
+    }
+
+    @Override
+    protected void postShutdownHook() {
+      try {
+        super.postShutdownHook();
+      } catch (Exception e) {
+        postShutdownHookExceptionCaught = true;
+      }
+    }
+
+    boolean isPostShutdownHookExceptionCaught() {
+      return postShutdownHookExceptionCaught;
+    }
+  }
+
+  @Test
+  public void testPartitionManagedLockReleaseOnInterruptException() throws InterruptedException {
+    Datastream datastream = KafkaMirrorMakerConnectorTestUtils.createDatastream("pizzaStream", _broker, "\\w+Pizza");
+    DatastreamTaskImpl task = new DatastreamTaskImpl(Collections.singletonList(datastream));
+    DatastreamEventProducer mockDatastreamEventProducer = mock(DatastreamEventProducer.class);
+    doThrow(InterruptException.class).when(mockDatastreamEventProducer).flush();
+    doNothing().when(mockDatastreamEventProducer).send(any(DatastreamProducerRecord.class), any(SendCallback.class));
```

Comment:
Did you need this line? I think doing nothing is Mockito's default behavior (unless otherwise instructed)

Reply:
done